src/LiteCore/src/LiteCore.Shared/Interop/Fleece.cs

```diff
@@ -391,14 +391,26 @@ public static void FLEncode(this object obj, FLEncoder* enc)
                 case string s:
                     s.FLEncode(enc);
                     break;
+                case byte b:
+                    ((ulong)b).FLEncode(enc);
+                    break;
+                case sbyte sb:
+                    ((long)sb).FLEncode(enc);
+                    break;
+                case ushort us:
+                    ((ulong)us).FLEncode(enc);
+                    break;
+                case short sh:
+                    ((long)sh).FLEncode(enc);
+                    break;
                 case uint u:
-                    ((ulong) u).FLEncode(enc);
+                    ((ulong)u).FLEncode(enc);
                     break;
                 case ulong u:
                     u.FLEncode(enc);
                     break;
                 case int i:
-                    ((long) i).FLEncode(enc);
+                    ((long)i).FLEncode(enc);
```

Comment:
You can simplify these cases in two groups by doing the following:

```
case byte b:
case ushort us:
case uint ui:
case ulong ul:
    var numericVal = Convert.ToUInt64(obj);
    numericVal.FLEncode(enc);
```

And the signed version